packages/astro/package.json

```diff
@@ -12,19 +12,19 @@
   },
   "bugs": "https://github.com/withastro/astro/issues",
   "homepage": "https://astro.build",
-  "types": "./dist/types/@types/astro.d.ts",
+  "types": "./dist/@types/astro.d.ts",
   "typesVersions": {
     "*": {
       "app": [
-        "./dist/types/core/app/index"
+        "./dist/core/app/index"
       ],
       "app/*": [
-        "./dist/types/core/app/*"
+        "./dist/core/app/*"
       ]
     }
   },
   "exports": {
-    ".": "./astro.js",
+    ".": {"default":"./astro.js", "types": "./dist/@types/astro.d.ts"},
```

Comment:
Note: the `types` condition should come before `default`. ([typescript docs](https://www.typescriptlang.org/docs/handbook/esm-node.html#packagejson-exports-imports-and-self-referencing))

Reply:
This is JSON, the order is arbitrary. I see the example in the docs has it before, but that's just an example, and nowhere in the text I read that it should come before (not as far as I can understand at least) - if I'm reading this wrong, please point out where I'm misreading.

Also do note that this is actually working, as I'm running locally with a `pnpm patch` to make astro work for me.

Reply:
It's shown in this code example in the linked docs:

<img width="747" alt="image" src="https://user-images.githubusercontent.com/34116392/183390424-c0b6327f-98dd-4516-95ce-40aa06e465ea.png">

and in the [announcement post](https://devblogs.microsoft.com/typescript/announcing-typescript-4-7/#package-json-exports-imports-and-self-referencing):

<img width="587" alt="image" src="https://user-images.githubusercontent.com/34116392/183390690-5c948719-7a90-408d-8198-10a1231912c0.png">

Disclaimer: I've created [publint](https://publint.bjornlu.com), so I'm picky with this 🙂

Reply:
It doesn't really matter in this case, but Node's documentation also mention that `default` should always be last

![Screenshot_20220808-075052.jpg](https://user-images.githubusercontent.com/3019731/183411951-88404781-0947-4c31-9a2e-34b8be9ec122.jpg)

Reply:
I love being picky as long as there's clear rules, and in this case there seems to be. I think it's a bad idea to have order-dependent JSON, so I'd argue node made a bad decision here, but it's what we got to live with, I guess. I'll update the pr when I'm back at my computer.

Reply:
Oh wow, I didn't know about these rules! That's super annoying, but yeah it would be great to make sure we're following them exactly.

Reply:
I have updated the order.